### PR TITLE
MASSEMBLY-922 - allow to override UID/GID for files stored in TAR

### PR DESCRIPTION
**Pull request by wilx**
This implements properties for assembly configuration that allow overriding UID/GID of files in TAR archive and any other archive for which plexus-archiver supports overriding of UID/GID.

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MASSEMBLY-922) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MASSEMBLY-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MASSEMBLY-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/assembly/mojos/AbstractAssemblyMojo.java [hunk 1]
```diff
@@ -404,6 +404,32 @@
     @Parameter( defaultValue = "${project.build.outputTimestamp}" )
     private String outputTimestamp;
 
+    /**
+     * Override of user ID in archive type which can store it.
+     */
+    @Parameter( property = "assembly.overrideUid" )
```
Comment (eolivelli) [inline src/main/java/org/apache/maven/plugins/assembly/mojos/AbstractAssemblyMojo.java]: Maybe it is better to name it maven.assembly.overrideUid ?

Cc @rfscholte

Comment (rfscholte) [reply src/main/java/org/apache/maven/plugins/assembly/mojos/AbstractAssemblyMojo.java]: Based on https://codehaus-plexus.github.io/plexus-archiver/apidocs/index.html?org/codehaus/plexus/archiver/Archiver.html it makes sense.

Comment (wilx) [reply src/main/java/org/apache/maven/plugins/assembly/mojos/AbstractAssemblyMojo.java]: All the other properties are just `assembly.*`.

Comment (rfscholte) [reply src/main/java/org/apache/maven/plugins/assembly/mojos/AbstractAssemblyMojo.java]: Ah, another question: does it make sense to set this via the commandline (`-Dkey=value`). I don't think so, or in other words: just use `@Parameter`

Comment (wilx) [reply src/main/java/org/apache/maven/plugins/assembly/mojos/AbstractAssemblyMojo.java]: IMHO, it seems nice to be able to do it via command line.

Comment (rfscholte) [reply src/main/java/org/apache/maven/plugins/assembly/mojos/AbstractAssemblyMojo.java]: Is there really a case the you when you need to change it for just one build? To me you want a consistent output, so hence not expose the parameter not with a property.
I noticed that the previous developers were quite enthousiastic with providing properties, but nowadays we're looking more closely if it really makes sense.

Comment (wilx) [reply src/main/java/org/apache/maven/plugins/assembly/mojos/AbstractAssemblyMojo.java]: I have removed the property names.

**Pull request merged**